src/librustc_codegen_llvm/context.rs

```diff
@@ -95,13 +97,18 @@ pub struct CodegenCx<'a, 'tcx: 'a> {
     pub pointee_infos: RefCell<FxHashMap<(Ty<'tcx>, Size), Option<PointeeInfo>>>,
     pub isize_ty: Type,
 
+    pub alloca_i8p_ty: Type,
+    pub const_i8p_ty: Type,
+    pub global_i8p_ty: Type,
+    pub flat_i8p_ty: Type,
+
     pub dbg_cx: Option<debuginfo::CrateDebugContext<'tcx>>,
 
     eh_personality: Cell<Option<ValueRef>>,
     eh_unwind_resume: Cell<Option<ValueRef>>,
     pub rust_try_fn: Cell<Option<ValueRef>>,
 
-    intrinsics: RefCell<FxHashMap<&'static str, ValueRef>>,
+    intrinsics: RefCell<FxHashMap<Cow<'static, str>, ValueRef>>,
```

Comment:
I think this is worse, because it incurs extra branching. Might as well switch it to `String`.